### PR TITLE
[Effects] Migrating data + a couple of functions

### PR DESCRIPTION
**Pull request by mrshigure**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 9 functions (+0.20%) on `us`.

File: src/effects/shockwave.c [hunk 1]
```diff
@@ -1,14 +1,255 @@
 #include "common.h"
 #include "effects_internal.h"
 
+extern Gfx D_090009A0_359D50[];
+extern Gfx D_09000A80_359E30[];
+
+Gfx* D_E004AE60[] = {
```
Comment (ethteck) [inline src/effects/shockwave.c:7]: these last two NULLs are probably data padding and can be removed

File: src/effects/red_impact.c [hunk 2]
```diff
@@ -1,14 +1,131 @@
 #include "common.h"
 #include "effects_internal.h"
 
+extern Gfx D_09000940_359CF0[];
+extern Gfx D_090009A0_359D50[];
+
+Gfx* D_E00D0B10[] = { D_090009A0_359D50, D_090009A0_359D50, NULL, NULL };
```
Comment (ethteck) [inline src/effects/red_impact.c:7]: same hee

File: src/effects/green_impact.c [hunk 3]
```diff
@@ -2,7 +2,9 @@
 #include "effects_internal.h"
 
 extern Gfx D_09000540_365EF0[];
-extern Gfx* D_E0064A60[];
+extern Gfx D_090005A0_365F50[];
+
+Gfx* D_E0064A60[] = { D_090005A0_365F50, D_090005A0_365F50, NULL, NULL };
```
Comment (ethteck) [inline src/effects/green_impact.c:7]: hi2

File: src/effects/gather_magic.c [hunk 4]
```diff
@@ -2,7 +2,10 @@
 #include "effects_internal.h"
 
 extern Gfx D_09000080_38F790[];
-extern Gfx* D_E008E890[];
+extern Gfx D_090001A8_38F8B8[];
+extern Gfx D_090001C8_38F8D8[];
+
+Gfx* D_E008E890[] = { D_090001A8_38F8B8, D_090001C8_38F8D8, NULL, NULL };
```
Comment (ethteck) [inline src/effects/gather_magic.c:8]: this2

File: src/effects/floating_rock.c [hunk 5]
```diff
@@ -1,8 +1,15 @@
...
 #include "effects_internal.h"
 
-extern Gfx D_09000450[];
-extern Gfx* D_E0098510[];
+extern Gfx D_090003A0_396490[];
+extern Gfx D_090003E0_3964D0[];
+extern Gfx D_09000410_396500[];
+extern Gfx D_09000450_396540[];
+
+Gfx* D_E0098510[] = {
```
Comment (ethteck) [inline src/effects/floating_rock.c:9]: zoinks scoob

File: src/effects/explosion.c [hunk 6]
```diff
@@ -4,7 +4,13 @@
 extern Gfx D_09000840_3447B0[];
 extern Gfx D_090008F0_344860[];
 extern Gfx D_09000910_344880[];
-extern Gfx* D_E00328B0[];
+extern Gfx D_09000978_3448E8[];
+extern Gfx D_090009D8_344948[];
+extern Gfx D_09000A38_3449A8[];
+
+Gfx* D_E00328B0[] = {
```
Comment (ethteck) [inline src/effects/explosion.c:11]: yowza

**Review by ethteck — CHANGES_REQUESTED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**